controller/cmd/service-mirror/cluster_watcher.go

```diff
@@ -912,7 +912,7 @@ func (rcsw *RemoteClusterServiceWatcher) affectedMirroredServicesForGatewayUpdat
 	affectedServices := []*corev1.Service{}
 	for _, srv := range services {
 		ver, ok := srv.Annotations[consts.RemoteGatewayResourceVersionAnnotation]
-		if ok && ver != latestResourceVersion {
+		if !ok || ver != latestResourceVersion {
```

Comment:
why this change ?

Reply:
It's unrelated to this change, but I believe it's a bug that was blocking my testing.

I had a situation where a mirror service was created before the gateway service (or during a period where the gateway service had been deleted).  This caused the mirror service to be created without the `RemoteGatewayResourceVersionAnnotation`.  When the gateway service was later created, this event did not trigger an update on the mirror service because the mirror service did not have the annotation (`ok` was false here).  This seems like the wrong behavior.  If the mirror service is missing this annotation, it should accept all gateway update events instead of rejecting all gateway update events.